server/src/modules/task/task.controller.ts

```diff
@@ -1,4 +1,45 @@
-import { Controller } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Delete,
+  Param,
+  Put,
+} from '@nestjs/common';
+import { TaskService } from './task.service';
+import { Task } from './task.model';
+import { TaskDTO as TaskInterface } from './task.interface';
 
 @Controller('task')
-export class TaskController {}
+export class TaskController {
+  constructor(private taskService: TaskService) {}
+
+  @Get('all-task')
+  async findAll(): Promise<TaskInterface[]> {
+    return this.taskService.getAllTask();
+  }
+
+  @Get(':id')
+  async getTaskById(@Param('id') id: number): Promise<Task> {
+    return this.taskService.getTaskById(id);
+  }
+
+  @Post('create')
+  async createTask(@Body() taskData: Task): Promise<Task> {
+    return this.taskService.createTask(taskData);
+  }
+
+  @Put(':id')
+  async updateTask(
+    @Param('id') id: number,
+    @Body() taskData: Task,
```

Comment:
Since you're using `Task` for updating tasks, it should already have an `id` field right? Do you want to just use `taskData.id` as the ID?